bottlecap/src/lifecycle/invocation/context.rs

```diff
@@ -38,6 +40,56 @@ pub struct Context {
     /// tracing.
     ///
     pub extracted_span_context: Option<SpanContext>,
+    /// The [`AppSecContext`] for this invocation, if one exists.
+    appsec_context: Option<AppSecContext>,
+}
+impl Context {
+    pub fn absorb_appsec_tags(&mut self) {
+        let Some(appsec_context) = self.appsec_context.as_mut() else {
+            // Nothing to do -- we don't have AppSec context data for this invocation.
+            return;
+        };
+
+        self.invocation_span
+            .metrics
+            .insert("_dd.appsec.enabled".to_string(), 1f64);
+
+        // Up-sert any attributes from the AppSec context into the invocation span. This includes
+        // synthetic attributes produced by the WAF (fingerprints, schemas, ...) as well as tracking
+        // spans expected by AAP. We do upsert to avoid overwriting anything previously set.
+        for (key, value) in appsec_context.tags() {
+            self.invocation_span
+                .meta
+                .entry(key.clone())
+                .or_insert(value.clone());
+        }
+
+        if !appsec_context.events.is_empty() {
+            self.invocation_span
+                .meta
+                .insert("appsec.event".to_string(), "true".to_string());
+            match serde_json::to_string(&appsec_context.events) {
+                Ok(encoded) => {
+                    self.invocation_span
+                        .meta
+                        .insert("_dd.appsec.json".to_string(), encoded);
+                }
+                Err(e) => {
+                    warn!("appsec: unable to encode WAF events: {e}");
+                }
+            }
+        }
+
+        // Note: We intentionally don't set `actor.ip` because we don't have a definitive signal here.
+
+        //TODO(romain.marcadier): Figure out whether we can set "_dd.runtime_family" here; and to what value.
```

Comment:
if there is a set of valid values, we can talk about how and when to set it, we get the specific runtime at a later point of init, but we guarantee it will arrive

Reply:
I see, we can do this at a later stage, but it should be possible as we get the set of runtimes with specific keys, we can just map them to the family later